opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -880,6 +882,9 @@ namespace Opm
         {
             const auto current = ws.production_cmode;
             const auto& controls = well.productionControls(summaryState);
+
+            ws.bhp = controls.bhp_limit;
+
```

Comment:
If your well is rate controlled the bhp_limit can be very far from the solution. What about just scaling it as we do with the rate. ws.bhp *= controls.oil_rate / current_rate. If the rates are close enough that is a good approximation.

Reply:
>  ws.bhp *= controls.oil_rate / current_rate. 

That will also be far, because the one related (proportionally) to the rates are drawdown, which is related to the difference between the wellbore pressure and the reservoir pressure. 

But thanks for the tips.